Amazon.ElastiCacheCluster/Pools/AutoServerPool.cs

```diff
@@ -299,13 +299,35 @@ void IDisposable.Dispose()
         public void UpdateLocator(List<IPEndPoint> endPoints)
         {
             var newLocator = this.configuration.CreateNodeLocator();
-            newLocator.Initialize(endPoints.Select(ip =>
+
+            var nodes = endPoints.Select(ip =>
             {
                 var node = this.CreateNode(ip);
                 node.Failed += this.NodeFail;
 
                 return node;
-            }).ToArray());
+            }).ToArray();
+
+            var aliveList = new List<IMemcachedNode>(nodes.Length);
+            var deadList = new List<IMemcachedNode>(nodes.Length);
+            foreach (var node in nodes)
+            {
+                var result = this.allNodes.Where(n => n.EndPoint.Equals(node.EndPoint)).ToList();
+
+                if (result.Count > 0 && !result[0].IsAlive)
+                {
+                    deadList.Add(result[0]);
+                    continue;
+                }
+
+                aliveList.Add(node);
+            }
+
+            newLocator.Initialize(aliveList);
+
+            // Retain All Nodes List With IsAlive Status
+            aliveList.AddRange(deadList);
```

Comment:
Shouldn't aliveLIst and deadList be combined into a separate collection. I assume NodeLocator is creating a separate list today when the aliveList is passed into Initialize which is why it is working now but I worry that might change sometime in the future and cause a painful to diagnose bug.